cudax/examples/cccl-rt/kernel_launch_patterns.cu

```diff
@@ -0,0 +1,254 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/devices>
+#include <cuda/std/cassert>
+#include <cuda/std/cstddef>
+#include <cuda/std/cstring>
+#include <cuda/stream>
+
+#include <cuda/experimental/hierarchy.cuh>
+#include <cuda/experimental/kernel.cuh>
+#include <cuda/experimental/launch.cuh>
+
+#include <cstdio>
+#include <stdexcept>
+
+#include <cuda.h>
+
+// Create an alias for the experimental namespace to shorten the code.
+namespace cudax = cuda::experimental;
+
+// A helper type for storing kernel launch patter name.
+struct name_buffer
```

Comment:
suggestion: I get the motivation for this, but I believe in an example any extra noise should be eliminated as much as possible to avoid distracting from the thing we are trying to teach. I would suggest getting rid of the `name_buffer` thing.